map-structure/vi-map/src/vi-map.cc

```diff
@@ -3084,15 +3093,21 @@ bool VIMap::mergeAllSubmapsFromMapWithoutResources(
         // If this landmark has been remapped to a landmark in the base map we
         // can delete it or if it doesn't have any observations. Otherwise we
         // need to update the global landmark index of the base map.
+        const bool has_observations = vertex_copy->getLandmarks()
+                                          .getLandmark(owned_landmark_id)
+                                          .hasObservations();
         if (remapped_landmark_submap_ids.count(owned_landmark_id) > 0u ||
-            !vertex_copy->getLandmarks()
-                 .getLandmark(owned_landmark_id)
-                 .hasObservations()) {
+            !has_observations) {
           vertex_copy->getLandmarks().removeLandmark(owned_landmark_id);
-          LandmarkId invalid;
-          vertex_copy->updateIdInObservedLandmarkIdList(
-              owned_landmark_id, invalid);
-        } else {
+          if (!has_observations) {
```

Comment:
even though it is removed we'd set it to invalid?

Reply:
yes, in the lines above we only remove it from the owned landmarks of this vertex, but we we still have to update the observation accordingly